src/number_theory/bernoulli_polynomials.lean

```diff
@@ -90,6 +91,25 @@ end
 
 end examples
 
+lemma deriv_bernoulli (k : ℕ) : (bernoulli k).derivative = k * bernoulli (k - 1) :=
+begin
+  rcases nat.eq_zero_or_pos k with hk|hk,
+  { rw [hk, nat.cast_zero, zero_mul, bernoulli_zero, derivative_one], },
+  simp_rw [bernoulli, derivative_sum, nat.sub_add_cancel hk, derivative_monomial],
+  -- LHS sum has an extra term, but the coefficient is zero:
+  rw [range_add_one, sum_insert not_mem_range_self, tsub_self, cast_zero, mul_zero, map_zero,
+    zero_add, mul_sum],
+  -- the rest of the sum is termwise equal:
+  refine sum_congr (by refl) (λ m hm, _),
+  rw [←C_eq_nat_cast, C_mul_monomial, nat.sub_sub, nat.sub_sub, add_comm],
+  congr' 1,
+  rw ←mul_assoc, conv { to_rhs, congr, rw mul_comm }, rw [mul_assoc, mul_assoc],
+  congr' 1, norm_cast,
+  convert (choose_mul_succ_eq (k - 1) m).symm using 1,
+  { rw nat.sub_add_cancel hk,},
+  { rw [mul_comm, nat.sub_add_cancel hk], },
+end
```

Comment:
Can you split out a `derivative_bernoulli_succ : (bernoulli (k + 1)).derivative = k + 1 * bernoulli k` lemma? I think it will make the proof clearer.